File: JovenVision.Domain/Entities/Attendance.cs
namespace JovenVision.Domain.Entities
{
    public class Attendance
    {
        public int Id { get; set; }
        public string Status { get; set; }
        public DateTime RegisteredAt { get; set; } = DateTime.Now;
        public int MemberId { get; set; }
        public int EventId { get; set; }
        public virtual Member Member { get; set; } = null!;
        public virtual Event Event { get; set; } = null!;
    }
}

File: JovenVision.Web/src/types/attendance.ts
import type { Member } from './member';
import type { JovenEvent } from './event';

export type AttendanceStatus = 'Present' | 'Absent' | 'Justified';

export interface Attendance {
  id: number;
  memberId: number;
  member?: Member;
  eventId: number;
  event?: JovenEvent;
  status: AttendanceStatus;
  registeredAt: string;
}

export interface AttendancePayload {
  memberId: number;
  eventId: number;
  status: AttendanceStatus;
}

export interface AttendanceSummary {
  total: number;
  present: number;
  absent: number;
  justified: number;
  percentage: number;
}
